scripts/install_backend_python_libs.py

```diff
@@ -42,6 +43,24 @@ def _normalize_python_library_name(library_name):
     Returns:
         str. A normalized library name that is all lowercase.
     """
+    # There are certain python libraries that have added support for other
+    # libraries. For example, 'google-api-core[grpc]' means that a variant of
+    # the 'google-api-core' package is installed that supports grpc. However,
+    # the import names, the package directory names, and the metadata directory
+    # names remain the same (google-api-core). These variant libraries are
+    # specified in the 'requirements.txt' file with the string,
+    # `library[sub-library]`. However, the installed version of this library
+    # is still considered by pkg_resources as having the default name `library`.
+    # This scenario causes this script to find incorrect mismatches. To solve
+    # this problem, we remove the special support package designation
+    # (e.g [grpc]) in the brackets when parsing the requirements file.
```

Comment:
Is it possible for multiple libraries to be listed in brackets? Also, what's the correct format for a library name? Can they never contain special characters (like brackets)?

Reply:
But if we add a dependency or a dependency updates in the future with multiple items in the square brackets, won't the implementation fail? I think we should make it robust against those failures.